File: web/src/app/models/game/move.ts
export class Move {
    source: number;
    destination: number;
    type: string;
    moveLog: string;

    constructor(source: number, destination: number, type: string, moveLog: string) {
        this.source = source;
        this.destination = destination;
        this.type = type;
        this.moveLog = moveLog;
    }
}

File: back/src/main/java/com/chess/spring/engine/moves/simple/Move.java
package com.chess.spring.engine.moves.simple;

import com.chess.spring.engine.board.PieceConfiguration;
import com.chess.spring.engine.pieces.utils.PieceGenre;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@AllArgsConstructor
@EqualsAndHashCode
public class Move {
    private int source;
    private int destination;
    private PieceGenre type;

    @Override
    public String toString() {
        return PieceConfiguration.getPositionAtCoordinate(this.source) + "-" + PieceConfiguration.getPositionAtCoordinate(this.destination);
    }
}
